### PR TITLE
dev: add clj-kondo source code linting

### LINKED ISSUE #28: dev cleanup: lint
## Currently

As of this writing, there are minor lint warnings from a default clj-kondo pass.

```
❯ clj-kondo --lint src test bb deps.edn bb.edn
src/clj_http/lite/core.clj:6:27: warning: Unused import SecureRandom
src/clj_http/lite/core.clj:85:25: warning: unused binding debug
test/clj_http/test/client.clj:3:50: warning: #'clj-http.test.core/current-port is referred but never used
test/clj_http/test/client.clj:7:23: warning: Unused import Arrays
test/clj_http/test/client.clj:75:21: warning: unused binding req
test/clj_http/test/client.clj:81:21: warning: unused binding req
test/clj_http/test/client.clj:87:21: warning: unused binding req
test/clj_http/test/client.clj:113:50: warning: unused binding req
test/clj_http/test/client.clj:145:21: warning: unused binding req
test/clj_http/test/client.clj:151:21: warning: unused binding req
test/clj_http/test/client.clj:158:21: warning: unused binding req
test/clj_http/test/client.clj:162:21: warning: unused binding req
test/clj_http/test/cookies.clj:2:4: warning: use :require with alias or :refer
test/clj_http/test/core.clj:4:14: warning: namespace clojure.pprint is required but never used
test/clj_http/test/core.clj:5:14: warning: namespace clojure.java.io is required but never used
test/clj_http/test/core.clj:5:38: warning: #'clojure.java.io/file is referred but never used
test/clj_http/test/core.clj:8:21: warning: Unused import ByteArrayInputStream
test/setup.clj:4:5: warning: namespace clojure.string is required but never used
test/setup.clj:9:29: warning: unused binding thread
linting took 220ms, errors: 0, warnings: 19
```

## Opinion
To me, this is noise that does not help me to quickly understand if I've introduced new potential issues with my changes.

## Proposal
Make a pass to clean up these warnings.

## Proposal++
Add a bb `lint` task.

Fail CI build on lint errors/warnings.

This one is a personal preference of mine.
I find it helps with code quality.

## Next Steps
If no one objects, I shall proceed with proposals.

### PR DESCRIPTION
**Pull request by lread**
Includes the addressing of reported lint findings.

Closes #28

### CODE REVIEW

**Comment by lread** [comment]
@borkdude this task is mostly stolen from my other projects... thought you might be interested in reviewing it before I merge since I hear you might know a thing or two about clj-kondo. 😉

File: bb/lint.clj [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+(defn- cache-exists? []
+  (fs/exists? clj-kondo-cache))
+
+(defn- delete-cache []
+  (when (cache-exists?)
+    (fs/delete-tree clj-kondo-cache)))
+
+(defn- build-cache []
+  (when (cache-exists?)
```
Comment (borkdude) [inline bb/lint.clj:19]: Why this?

**Comment by borkdude** [comment]
@lread To my taste, it's a little bit boilerplatey but maybe for good reasons. What I would do personally is just lint the project sources without any of the caching/pre-setup stuff, using setup-clojure's clj-kondo install: just a few lines in the Github actions config. But I'm fine with your approach too, it's just more maintenance-sensitive and it may give others the impression that setting up clj-kondo for your project is a verbose task.

**Comment by lread** [comment]
Thanks for the feedback, very reasonable observations.

Here are some of the things I really like about my current linting approach:
1. having a bb task highlights linting as important
2. allows devs to lint the very same way as CI does
3. brings in all bb and clj deps and their exported configs (and automatically redoes this when deps change)
4. allows anyone to easily force step 3 via --rebuild

I'd like to stick with this approach for now, unless any of the above gives you new concerns.

**Comment by borkdude** [comment]
LGTM

**Pull request merged**